loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+            parent_id = self._id_map[str(id(parent))]
+            child_id = self._id_map[str(id(node))]
+            edge_info = {"tail_name": str(parent_id), "head_name": str(child_id)}
+
+        return [(node_info, edge_info)]
+
+    # Handler for outer objects
+    def visit_Module(self, o, **kwargs):
+        """
+        Add a :any:`Module`, mark parent node and visit all "spec" and "subroutine" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.subroutines, **kwargs))
+
+        return node_edge_info
+
+    def visit_Subroutine(self, o, **kwargs):
+        """
+        Add a :any:`Subroutine`, mark parent node and visit all "docstring", "spec", "body", "members" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.docstring, **kwargs))
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.body, **kwargs))
+        node_edge_info.extend(self.visit(o.members, **kwargs))
+
+        return node_edge_info
+
+    # Handler for AST base nodes
+    def visit_Comment(self, o, **kwargs):
+        """
+        Enables turning off comments.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information, or list of nothing.
+        """
+        if kwargs.get("show_comments"):
+            return self.visit_Node(o, **kwargs)
+        return [None]
+
+    visit_CommentBlock = visit_Comment
+
+    def visit_Node(self, o, **kwargs):
+        """
+        Add a :any:`Node`, mark parent and visit all children.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit_all(o.children, **kwargs))
+        return node_edge_info
+
+    def visit_Expression(self, o, **kwargs):
+        """
+        Dispatch routine to add nodes utilizing expression tree stringifier,
+        mark parent and stop.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information or list of nothing.
+        """
+        if kwargs.get("show_expressions"):
```

Comment:
Same as for `show_comments`: This is behaviour parameterisation for the entire class - hence I would suggest to make this an object attribute and to add it to the constructor instead of passing it around via kwargs.